### PR TITLE
Upstream the Guidebook

### PR DESCRIPTION
**Pull request by moonheart08**
image

**Changelog**
:cl: moony
- add: An in-game guide has been added that'll hopefully ease on-boarding for new players. Help wanted with writing!

### CODE REVIEW

**Comment by moonheart08** [comment]
okay this git history looks dirty but it's also all stuff that got merged upstream so it ultimately contributes nothing.

**Comment by moonheart08** [comment]
Undrafting as i'll never be happy with the code behind this unless i rewrite all of richtext

**Review by vulppine — CHANGES_REQUESTED** [review]
Should the tutorial in `Content.Client/Info/RulesAndInfoWindow.cs` be removed as a result of this? It seems redundant, now.

File: Content.Client/Guidebook/GuidebookWindow.Layout.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            }
+            case "controlsButton":
+            {
+                var button = new Button()
+                {
+                    Text = Loc.GetString("ui-info-button-controls"),
+                };
+                button.OnPressed += _ => new OptionsMenu().Open();
+                control = button;
+                return;
```
Comment (ElectroJr) [inline Content.Client/Guidebook/GuidebookWindow.Layout.xaml.cs:186]: Hard-coding a directive to open a specific button, which will probably only ever be used in a single guidebook feels very icky.

Maybe make it a general button to run client-side commands, and create a `openoptions` command (if one doesn't already exist?)

**Review by ElectroJr — CHANGES_REQUESTED** [review]
I haven't actually looked at any of the guidebook text, other than making the funny disco monkey dance.

File: Content.Client/Guidebook/GuidebookWindow.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+
+    public void UpdateGuides(List<GuideEntry> entries)
+    {
+        _entries = entries;
+        RedrawTree();
+    }
+
+    private void RedrawTree()
+    {
```
[the remark this answers is not in the document]
Comment (moonheart08) [reply Content.Client/Guidebook/GuidebookWindow.xaml.cs:59]: I don't want to write a brand new tree UI element so this is what we've got.

File: Content.Client/Guidebook/GuidebookWindow.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    public GuidebookWindow()
+    {
+        RobustXamlLoader.Load(this);
+        IoCManager.InjectDependencies(this);
+
+        GuideSelect.OnItemSelected += GuideSelectOnOnItemSelected;
+    }
+
+    private void GuideSelectOnOnItemSelected()
+    {
```
[the remark this answers is not in the document]
Comment (moonheart08) [reply Content.Client/Guidebook/GuidebookWindow.xaml.cs:26]: There is no function to do this on the tree UI element.

File: Content.Client/Guidebook/GuidebookWindow.Layout.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+                    {
+                        layoutMode = LayoutMode.Header;
+                        newLine = false;
+                        continue;
+                    }
+
+                    /*case var ws when Char.IsWhiteSpace(ws):
+                    {
+                        continue; // Keep in NL mode.
+                    }*/
```
[the remark this answers is not in the document]
Comment (moonheart08) [reply Content.Client/Guidebook/GuidebookWindow.Layout.xaml.cs:41]: tl;dr it does not work right and making it work right requires a parser rewrite.

**Pull request closed without merging**